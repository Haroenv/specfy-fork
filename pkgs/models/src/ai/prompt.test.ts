import { describe, expect, it } from 'vitest';

import {
  aiPromptProjectDescription,
  aiPromptRewrite,
  aiPromptProjectOnboarding,
} from './prompt.js';

describe('aiPromptRewrite', () => {
  it('should output the prompt', () => {
    const res = aiPromptRewrite({
      text: 'Specfy is an awesome platform',
    });
    expect(res).toMatchSnapshot();
  });
});

describe('aiPromptProjectDescription', () => {
  it('should output the prompt', () => {
    const res = aiPromptProjectDescription({
      project: { name: 'Analytics' },
      components: [
        { name: 'GCP', type: 'hosting', techId: 'gcp', slug: 'gcp' },
        { name: 'GCE', type: 'hosting', techId: 'gcp.gce', slug: 'gce' },
        { name: 'Aggregator', type: 'service', techId: null, slug: 'aggr' },
        { name: 'Algolia', type: 'saas', techId: 'algolia', slug: 'algolia' },
        { name: 'Postgres', type: 'db', techId: 'postgres', slug: 'postgres' },
        { name: 'SQS', type: 'messaging', techId: 'aws.sqs', slug: 'sqs' },
        { name: 'Billing', type: 'project', techId: null, slug: 'billing' },
      ],
    });
    expect(res).toMatchSnapshot();
  });
});

describe('aiPromptProjectOnboarding', () => {
  it('should output the prompt', () => {
    const res = aiPromptProjectOnboarding({
      project: {
        name: 'Analytics',
        slug: 'analytics',
        orgId: 'acme',
        links: [{ title: 'Github', url: 'https://github.com/specfy/specfy' }],
        description: {
          type: 'doc',
          content: [
            {
              type: 'paragraph',
              attrs: { uid: 'UidC3Ls190' },
              content: [
                {
                  type: 'text',
                  text: 'The Analytics project collects data from various sources, processes it, and presents it in dashboards and reports. These outputs show key indicators and trends for the business.',
                },
                { type: 'hardBreak' },
                {
                  type: 'text',
                  text: 'The project also employs machine learning and statistical methods to analyze data. This analysis helps in predicting trends and making recommendations based on past data. This project assists businesses in making data-based decisions.',
                },
              ],
            },
          ],
        },
      },
      components: [
        { name: 'GCP', type: 'hosting', techId: 'gcp', slug: 'gcp' },
        { name: 'GCE', type: 'hosting', techId: 'gcp.gce', slug: 'gce' },
        { name: 'Aggregator', type: 'service', techId: null, slug: 'aggr' },
        { name: 'Algolia', type: 'saas', techId: 'algolia', slug: 'algolia' },
        { name: 'Postgres', type: 'db', techId: 'postgres', slug: 'postgres' },
        { name: 'SQS', type: 'messaging', techId: 'aws.sqs', slug: 'sqs' },
        { name: 'Billing', type: 'project', techId: null, slug: 'billing' },
      ],
      documents: [
        { name: 'README', slug: 'readme' },
        { name: 'Contributing', slug: 'docs/contributing' },
        { name: 'Installation', slug: 'docs/installation' },
        { name: 'Deploy', slug: 'docs/deploy' },
        { name: 'Deploy With Terraform', slug: 'docs/deploy.terraform' },
      ],
    });
    expect(res).toMatchSnapshot();
  });
});
