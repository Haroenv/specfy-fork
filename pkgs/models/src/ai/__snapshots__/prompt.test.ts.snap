// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`aiPromptProjectDescription > should output the prompt 1`] = `
[
  {
    "content": "You describe a technical project for an engineer.
3 paragraphs maximum
DO NOT use adjective, superlative or adverb.
Details why it exists in one paragraph with a line break at end
Details briefly how it works
Use the list of technologies to describe how it works. Choose only the most relevant ones",
    "role": "system",
  },
  {
    "content": "Project name is Analytics

Technologies:
Hosting: GCP, GCE
Service: Aggregator
Saas: Algolia
Database: Postgres
Messaging: SQS
Depends on internal product: Billing
  ",
    "role": "user",
  },
]
`;

exports[`aiPromptProjectOnboarding > should output the prompt 1`] = `
[
  {
    "content": "You are technical documentation assistant, you are given a description, a list of files for the documentation, a list of technologies used in the project and a list of link.
Write 4 sections:
1. A general introduction for project titled Onboarding
2. A list of steps in english based on the documentation, only pick the relevant link that would be interesting for an onboarding with explanation. Select one relevant emoji per step.
Prepend the link with the url \\"http://localhost:5173/acme/analytics/doc/\\"
Following this pattern:
<li>{emoji} <strong>{name of the step}</strong>: {explanation} {link with the name}</li>
3. Highlight the most important technologies from the list and explain what they do in one sentence. Maximum 5.
Link this resource by prepending this url \\"http://localhost:5173/acme/analytics/c/\\" to the id
Following this pattern:
<li><strong>{name of the technology}</strong>: {explanation} {link with the name}</li>
4. Next steps in bullet point. Use the links if any.

Output in HTML without the body tag",
    "role": "system",
  },
  {
    "content": "Project name is Analytics
The Analytics project collects data from various sources, processes it, and presents it in dashboards and reports. These outputs show key indicators and trends for the business.
The project also employs machine learning and statistical methods to analyze data. This analysis helps in predicting trends and making recommendations based on past data. This project assists businesses in making data-based decisions.



Technologies:
Hosting: GCP (id: gcp), GCE (id: gce)
Service: Aggregator (id: aggr)
Saas: Algolia (id: algolia)
Database: Postgres (id: postgres)
Messaging: SQS (id: sqs)
Depends on internal product: Billing (id: billing)

Documentation:
/readme : README
/docs/contributing : Contributing
/docs/installation : Installation
/docs/deploy : Deploy
/docs/deploy.terraform : Deploy With Terraform

Links:
Github: https://github.com/specfy/specfy
Infrastructure Flow: http://localhost:5173/acme/analytics/flow
",
    "role": "user",
  },
]
`;

exports[`aiPromptRewrite > should output the prompt 1`] = `
[
  {
    "content": "You rewrite content you are provided with for an engineer. Do not use adverb or subjective language.",
    "role": "system",
  },
  {
    "content": "Specfy is an awesome platform",
    "role": "user",
  },
]
`;
